tests/net/lib/mqtt_pubsub/src/test_mqtt_pubsub.c

```diff
@@ -422,34 +422,37 @@ static int test_disconnect(void)
 	return TC_PASS;
 }
 
-void test_mqtt_connect(void)
+
+void test_mqtt_connect(void *dummy)
 {
+	ARG_UNUSED(dummy);
 	zassert_true(test_connect() == TC_PASS, NULL);
 }
 
-void test_mqtt_subscribe(void)
+ZTEST(net_mqtt_pubsub, test_mqtt_subscribe)
 {
 	zassert_true(test_subscribe() == TC_PASS, NULL);
 }
 
-void test_mqtt_publish_short(void)
+ZTEST(net_mqtt_pubsub, test_mqtt_publish_short)
```

Comment:
In this partucular case, specific test cases are dependend on each other, for example `test_mqtt_publish_short` expects that the client is already subscribed (`test_mqtt_subscribe` already ran). I think that for this test suite we should define a single test case with the new API, executing all off the steps in order.

Reply:
OK, updated.
Is this version what you want?

Reply:
yes, thank you